src/ESM.t.sol

```diff
@@ -224,76 +232,133 @@ contract ESMTest is DSTest {
+        gem.mint(address(usr), 10_000 * WAD);
+        esm = new ESM(address(gem), address(end), address(0), 10_000 * WAD);
+        usr.callJoin(esm, 10_000 * WAD);
+        assertEq(gem.balanceOf(address(esm)), 10_000 * WAD);
         usr.callBurn(esm);
         assertEq(gem.balanceOf(address(esm)), 0);
         usr.callFire(esm);
     }
 
     function test_burn_after_fire() public {
-        gem.mint(address(usr), 10);
-        esm = new ESM(address(gem), address(end), address(0), 10);
-        usr.callJoin(esm, 10);
-        assertEq(gem.balanceOf(address(esm)), 10);
+        gem.mint(address(usr), 10_000 * WAD);
+        esm = new ESM(address(gem), address(end), address(0), 10_000 * WAD);
+        usr.callJoin(esm, 10_000 * WAD);
+        assertEq(gem.balanceOf(address(esm)), 10_000 * WAD);
         usr.callFire(esm);
         usr.callBurn(esm);
         assertEq(gem.balanceOf(address(esm)), 0);
     }
 
     function test_join_over_min() public {
-        gem.mint(address(usr), 20);
-        esm = new ESM(address(gem), address(end), address(0), 10);
+        gem.mint(address(usr), 20_000 * WAD);
+        esm = new ESM(address(gem), address(end), address(0), 10_000 * WAD);
 
-        usr.callJoin(esm, 10);
-        usr.callJoin(esm, 10);
+        usr.callJoin(esm, 10_000 * WAD);
+        usr.callJoin(esm, 10_000 * WAD);
     }
 
     function testFail_join_insufficient_balance() public {
-        esm = new ESM(address(gem), address(end), address(0), 10);
+        esm = new ESM(address(gem), address(end), address(0), 10_000 * WAD);
         assertEq(gem.balanceOf(address(usr)), 0);
 
-        usr.callJoin(esm, 10);
+        usr.callJoin(esm, 10_000 * WAD);
+    }
+
+    function test_file_new_min() public {
+        esm = new ESM(address(gem), address(end), address(this), 10_000 * WAD);
+        assertEq(esm.min(), 10_000 * WAD);
+
+        esm.file("min", 20_000 * WAD);
+
+        assertEq(esm.min(), 20_000 * WAD);
+    }
+
+    function test_file_new_min_then_fire() public {
+        gem.mint(address(usr), 10_000 * WAD);
+        esm = new ESM(address(gem), address(end), address(this), 20_000 * WAD);
+        vat.rely(address(esm));
+
+        assertEq(esm.min(), 20_000 * WAD);
+        assertEq(vat.wards(address(this)), 1);
+
+        usr.callJoin(esm, 10_000 * WAD);
+        esm.file("min", 10_000 * WAD);
+
+        assertEq(esm.min(), 10_000 * WAD);
+
+        usr.callFire(esm);
+
+        assertEq(vat.wards(address(this)), 0);
+        assertEq(end.live(), 0);
+    }
+
+    function testFail_file_revoked_gov() public {
+        esm = new ESM(address(gem), address(end), address(0), 10_000 * WAD);
+        esm.deny(address(this));
+        assertEq(esm.min(), 10_000 * WAD);
+
+        esm.file("min", 20_000 * WAD);
+    }
+
+    function testFail_file_not_gov() public {
+        esm = new ESM(address(gem), address(end), pauseProxy, 10_000 * WAD);
+        assertEq(esm.min(), 10_000 * WAD);
+
```

Comment:
```suggestion
        esm.rely(address(usr));
```

Reply:
I think I misplaced this one

Reply:
nevermind, it is good

Reply:
i updated the title of this one, but since it is testing that not-relied addresses can't file, i don't think we should have this.

Reply:
I added another test explicitly testing the happy path for these actions since that is our primary path with the deployer/gov

Reply:
yes, I probably got confused